File: frontend/src/domains/Score.ts
import { AttendSubject, Evaluation } from '.';

export class Score {
  constructor(
    public uuid: string,
    public attendSubject: AttendSubject,
    public evaluation: Evaluation,
    public got_score: number,
    public max_score: number,
    public memo: string,
  ) {}
}

export class ScoreEval {
  constructor(
    public score_uuid: string,
    public attendSubject: AttendSubject,
    public evaluation: Evaluation,
    public got_score: number,
    public max_score: number,
    public memo: string,
  ) {}
}

File: fastapi/app/schemas/score.py
from typing import Optional
from uuid import UUID

from app.models import Score
from pydantic import BaseModel, Field

from .evaluation import ReadEvaluationSchema


class BaseScoreSchema(BaseModel):

    got_score: int
    max_score: int
    memo: Optional[str] = Field(None, max_length=Score.MAX_LENGTH_MEMO)

    class Config:
        orm_mode = True


class ReadScoreSchema(BaseScoreSchema):
    uuid: UUID
    evaluation: ReadEvaluationSchema


class CreateScoreSchema(BaseScoreSchema):
    attend_subject_uuid: UUID
    evaluation_uuid: UUID


class UpdateScoreSchema(BaseScoreSchema):
    attend_subject_uuid: UUID
    evaluation_uuid: UUID


class ReadSimpleScoreSchema(BaseScoreSchema):
    uuid: UUID
